### PR TITLE
[pages] Add Cloudflare Tunnel guide

### LINKED ISSUE #1425: [pages] Add how-to guide on using Cloudflare Tunnel for local preview
@rita3ko's blog post on [Cloudflare Pages GA](https://blog.cloudflare.com/cloudflare-pages-ga/) showed off how to use Cloudflare Tunnel to do local development/preview for Pages projects before they get deployed. Since then, I've been using Tunnel for a lot of local development, and it's super popular in our Discord too - let's write a brief guide showing how to install Cloudflare Tunnel, and configure it for simple ngrok-style workflows: running a project like Gatsby or Next locally, and exposing port 3000 for other users to check out via a public tunnel URL.

### PR DESCRIPTION
**Pull request by kristianfreeman**
Preview URL: https://telecommunications-ph-apartments-dream.trycloudflare.com/how-to/preview-with-cloudflare-tunnel

Closes #1425

### CODE REVIEW

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Confirm that `cloudflared` is installed correctly by running `cloudflared --version` in your command-line:
+
+```sh
+$ cloudflared --version
+cloudflared version 2021.5.9 (built 2021-05-21-1541 UTC)
+```
+
+## Run a local service
+
+The easiest way to get up and running with Cloudflare Tunnel is to have an application running locally, such as a [React](/framework-guides/deploy-a-react-application) or [Svelte](/framework-guides/deploy-a-svelte-site) site. When you are developing an application with these frameworks, they will often make use of a `npm run develop` script, or something similar, which mounts the application and runs it on a `localhost` port. For instance, the popular `create-react-app` tool runs your in-development React application on port 3000, or `localhost:3000`.
```
Comment (lukeed) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
The easiest way to get up and running with Cloudflare Tunnel is to have an application running locally, such as a [React](/framework-guides/deploy-a-react-application) or [Svelte](/framework-guides/deploy-a-svelte-site) site. When you are developing an application with these frameworks, they will often make use of a `npm run develop` script, or something similar, which mounts the application and runs it on a `localhost` port. For instance, the popular `create-react-app` tool runs your in-development React application on port 3000, making it accessible at the `http://localhost:3000` address.
```

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+2021-07-15T20:11:32Z INF +-------------------------------------------------------------+
+```
+
+In this example, the randomly-generated URL `https://seasonal-deck-organisms-sf.trycloudflare.com` has been created and assigned to your tunnel instance. Visiting this URL in a browser will show the application running, with requests being securely forwarded through Cloudflare's edge network, through the tunnel running on your machine, to `localhost:3000`:
+
+![Cloudflare Tunnel example](./media/tunnel.png)
+
+## Next Steps
+
+Cloudflare Tunnel can be configured in a variety of ways, to provide more than just simple access to your in-development applications. For instance, you can pass `cloudflared` a [config file](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config) to add more complex routing and tunnel setups, and [use Cloudflare DNS](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/dns) to associate a domain or subdomain with a long-running tunnel on your machine.
```
Comment (lukeed) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
Cloudflare Tunnel can be configured in a variety of ways and can be used to provide far more than just simple access to your in-development applications. For instance, you can provide `cloudflared` with a [configuration file](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config) to add more complex routing and tunnel setups that go beyond a simple `--url` flag. You can also [attach a Cloudflare DNS record](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/dns) to a domain or subdomain for an easily-accessible, long-lived tunnel to your local machine. And with a [Cloudflare One](https://developers.cloudflare.com/cloudflare-one/) access policy in place, this means only you (or your teammates) can use `https://local.company.com`, for example.
```

Comment (lukeed) [reply products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]: Oh  – jk on the last sentence. I didn't see the next paragraph before writing it. 🤦

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+cloudflared version 2021.5.9 (built 2021-05-21-1541 UTC)
+```
+
+## Run a local service
+
+The easiest way to get up and running with Cloudflare Tunnel is to have an application running locally, such as a [React](/framework-guides/deploy-a-react-application) or [Svelte](/framework-guides/deploy-a-svelte-site) site. When you are developing an application with these frameworks, they will often make use of a `npm run develop` script, or something similar, which mounts the application and runs it on a `localhost` port. For instance, the popular `create-react-app` tool runs your in-development React application on port 3000, or `localhost:3000`.
+
+## Start a Cloudflare Tunnel
+
+With this development server running, a new Cloudflare Tunnel can be instantiated by running `cloudflared tunnel`, passing in the `--url` flag with your `localhost` URL and port. `cloudflared` will output logs to your command-line, including a banner with a tunnel URL:
```
Comment (rita3ko) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]: not sure if this is obvious to people or not, but you'll want to run the instance on a separate terminal tab / instance from where you're running localhost (e.g. if you `ctrl+c` out of `npm run develop` and then run your tunnel, that won't work.
maybe this is obvious

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
+---
+pcx-content-type: how-to
+---
+
+# Preview Local Projects with Cloudflare Tunnel
+
+[Cloudflare Tunnel](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps) runs a lightweight daemon (`cloudflared`) in your infrastructure that establishes outbound connections (Tunnels) between your service and the Cloudflare edge. In practical terms, you can use Cloudflare Tunnel to allow remote access to services running on your local machine. It is an alternative to popular tools like [Ngrok](https://ngrok.com), and provides free, long-running tunnels via the [TryCloudflare](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/trycloudflare) service.
```
Comment (deadlypants1973) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
[Cloudflare Tunnel](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps) runs a lightweight daemon (`cloudflared`) in your infrastructure that establishes outbound connections (Tunnels) between your origin web server and the Cloudflare edge. In practical terms, you can use Cloudflare Tunnel to allow remote access to services running on your local machine. It is an alternative to popular tools like [Ngrok](https://ngrok.com), and provides free, long-running tunnels via the [TryCloudflare](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/trycloudflare) service.
```

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+While Cloudflare Pages provides unique [deploy preview URLs](/platform/preview-deployments) for new branches and commits on your projects, Cloudflare Tunnel can be used to provide access to locally running applications and servers during the development process. In this guide, you will install Cloudflare Tunnel, and create a new tunnel to provide access to a locally running application. You will need a Cloudflare account to begin using Cloudflare Tunnel.
+
+## Installing Cloudflare Tunnel
+
+Cloudflare Tunnel can be installed on Windows, Linux, and macOS. To learn about how to install Cloudflare Tunnel, see the ["Install cloudflared"](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/installation) page in the Cloudflare for Teams documentation.
```
Comment (deadlypants1973) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
Cloudflare Tunnel can be installed on Windows, Linux, and macOS. To learn about installing Cloudflare Tunnel, refer to the ["Install cloudflared"](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/install-and-setup/installation) page in the Cloudflare for Teams documentation.
```

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+Confirm that `cloudflared` is installed correctly by running `cloudflared --version` in your command-line:
```
Comment (deadlypants1973) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
Confirm that `cloudflared` is installed correctly by running `cloudflared --version` in your command line:
```

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md (hunk 3 above)
Comment (deadlypants1973) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
With a local development server running, a new Cloudflare Tunnel can be instantiated by running `cloudflared tunnel`, passing in the `--url` flag with your `localhost` URL and port. `cloudflared` will output logs to your command line, including a banner with a tunnel URL:
```

Not sure if this is right but "this" should be defined for clarity.

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+![Cloudflare Tunnel example](./media/tunnel.png)
+
+## Next Steps
+
+Cloudflare Tunnel can be configured in a variety of ways, to provide more than just simple access to your in-development applications. For instance, you can pass `cloudflared` a [config file](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config) to add more complex routing and tunnel setups, and [use Cloudflare DNS](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/dns) to associate a domain or subdomain with a long-running tunnel on your machine. 
+
+In conjunction with Cloudflare Access, you can provide [secure access to your tunnels](https://developers.cloudflare.com/cloudflare-one/applications/configure-apps/self-hosted-apps) without exposing your entire server, or compromising on security. See the [Cloudflare for Teams documentation](https://developers.cloudflare.com/cloudflare-one/) to learn more about what you can do with our entire suite of Zero Trust tools.
```
Comment (deadlypants1973) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
In conjunction with Cloudflare Access, you can provide [secure access to your tunnels](https://developers.cloudflare.com/cloudflare-one/applications/configure-apps/self-hosted-apps) without exposing your entire server, or compromising on security. Refer to the [Cloudflare for Teams documentation](https://developers.cloudflare.com/cloudflare-one/) to learn more about what you can do with Cloudflare's entire suite of Zero Trust tools.
```

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+![Cloudflare Tunnel example](./media/tunnel.png)
+
+## Next Steps
+
+Cloudflare Tunnel can be configured in a variety of ways and can be used to provide far more than just simple access to your in-development applications. For instance, you can provide `cloudflared` with a [configuration file](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config) to add more complex routing and tunnel setups that go beyond a simple `--url` flag. You can also [attach a Cloudflare DNS record](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/dns) to a domain or subdomain for an easily-accessible, long-lived tunnel to your local machine.
+
+In conjunction with Cloudflare Access, you can provide [secure access to your tunnels](https://developers.cloudflare.com/cloudflare-one/applications/configure-apps/self-hosted-apps) without exposing your entire server, or compromising on security. Refer to the [Cloudflare for Teams documentation](https://developers.cloudflare.com/cloudflare-one/) to learn more about what you can do with Cloudflare's entire suite of Zero Trust tools.
```
Comment (lukeed) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
Finally, by incorporating Cloudflare Access, you can provide [secure access to your tunnels](https://developers.cloudflare.com/cloudflare-one/applications/configure-apps/self-hosted-apps) without exposing your entire server, or compromising on security. Refer to the [Cloudflare for Teams documentation](https://developers.cloudflare.com/cloudflare-one/) to learn more about what you can do with Cloudflare's entire suite of Zero Trust tools.
```

Comment (lukeed) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
In conjunction with Cloudflare Access, you can provide [secure access to your tunnels](https://developers.cloudflare.com/cloudflare-one/applications/configure-apps/self-hosted-apps) without exposing your entire server or compromising on its security. Please refer to the [Cloudflare for Teams documentation](https://developers.cloudflare.com/cloudflare-one/) to learn more about what you can do with Cloudflare's entire suite of Zero Trust tools.
```

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+Confirm that `cloudflared` is installed correctly by running `cloudflared --version` in your command line:
+
+```sh
+$ cloudflared --version
+cloudflared version 2021.5.9 (built 2021-05-21-1541 UTC)
+```
+
+## Run a local service
+
+The easiest way to get up and running with Cloudflare Tunnel is to have an application running locally, such as a [React](/framework-guides/deploy-a-react-application) or [Svelte](/framework-guides/deploy-a-svelte-site) site. When you are developing an application with these frameworks, they will often make use of a `npm run develop` script, or something similar, which mounts the application and runs it on a `localhost` port. For instance, the popular `create-react-app` tool runs your in-development React application on port 3000, making it accessible at the `http://localhost:3000` address.
```
Comment (deadlypants1973) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
The easiest way to get up and running with Cloudflare Tunnel is to have an application running locally, such as a [React](/framework-guides/deploy-a-react-application) or [Svelte](/framework-guides/deploy-a-svelte-site) site. When you are developing an application with these frameworks, they will often make use of a `npm run develop` script, or something similar, which mounts the application and runs it on a `localhost` port. For example, the popular `create-react-app` tool runs your in-development React application on port 3000, making it accessible at the `http://localhost:3000` address.
```

File: products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+2021-07-15T20:11:32Z INF +-------------------------------------------------------------+
+```
+
+In this example, the randomly-generated URL `https://seasonal-deck-organisms-sf.trycloudflare.com` has been created and assigned to your tunnel instance. Visiting this URL in a browser will show the application running, with requests being securely forwarded through Cloudflare's edge network, through the tunnel running on your machine, to `localhost:3000`:
+
+![Cloudflare Tunnel example](./media/tunnel.png)
+
+## Next Steps
+
+Cloudflare Tunnel can be configured in a variety of ways and can be used to provide far more than just simple access to your in-development applications. For instance, you can provide `cloudflared` with a [configuration file](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config) to add more complex routing and tunnel setups that go beyond a simple `--url` flag. You can also [attach a Cloudflare DNS record](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/dns) to a domain or subdomain for an easily-accessible, long-lived tunnel to your local machine.
```
Comment (deadlypants1973) [inline products/pages/src/content/how-to/preview-with-cloudflare-tunnel.md]:
```suggestion
Cloudflare Tunnel can be configured in a variety of ways and can be used beyond providing access to your in-development applications. For example, you can provide `cloudflared` with a [configuration file](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/configuration/config) to add more complex routing and tunnel setups that go beyond a simple `--url` flag. You can also [attach a Cloudflare DNS record](https://developers.cloudflare.com/cloudflare-one/connections/connect-apps/routing-to-tunnel/dns) to a domain or subdomain for an easily accessible, long-lived tunnel to your local machine.
```
No dash is needed after adverbs that end in -ly.

**Review by deadlypants1973 — APPROVED** [review]
Approved but let me know if you have any comments, @abracchi-tw

**Pull request merged**